### PR TITLE
Make it possible to disable BLS verification for everything except Deposits (attempt 2)

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Replace `AbstractBlockProcessor.blsVerifyDeposit` static flag with `BLSSignatureVerifier` reference to be able to set the verifier to the working instance even if the BLS verification is globally disabled with `BLSConstants.disableBLSVerification()`

This is the second attempt to enable this feature. 
The first one #6116 had an issue #6147 and had to be reverted: #6148

The problem was in inconsistent `BLS.verify*` methods behavior with respect to invalid `BLSPublicKey`: some methods returned `false`, some thrown `BLSException`. This PR also aligns these methods behavior: they are now all return `false`. Additionally `catch(BLSException)` clauses were added around `verify*` calls in `AbstractBlockProcessor` before (in revert PR #6148)

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by tbenr** [comment]
Today @rolfyone was actually working on the BLS verification area. You may need some coordination :)

**Comment by rolfyone** [comment]
> Today @rolfyone was actually working on the BLS verification area. You may need some coordination :)

I'll keep digging and see if theres reasons I can't do what I'm thinking, but I'd really like to be avoiding using BlsSignatureVerifier.SIMPLE, or somehow making sure we minimise individual verify calls where possible...

It's just that in production we seem to be spending a significant amount of time doing individual verifies rather than batching... I'll drill into uses of the single verify function when i get a sec...

**Review by rolfyone — APPROVED** [review]
LGTM

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java [hunk 1]
```diff
@@ -81,11 +81,16 @@
...
+  @VisibleForTesting
+  public static final BLSSignatureVerifier DEFAULT_DEPOSIT_SIGNATURE_VERIFIER =
+      BLSSignatureVerifier.SIMPLE;
   /**
    * For debug/test purposes only enables/disables {@link DepositData} BLS signature verification
    * Setting to <code>false</code> significantly speeds up state initialization
    */
-  public static boolean blsVerifyDeposit = true;
+  @VisibleForTesting
+  public static BLSSignatureVerifier depositSignatureVerifier = DEFAULT_DEPOSIT_SIGNATURE_VERIFIER;
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java:93]: are we planning on differentiating? why not just reference the default_deposit_signature_verifier?

Comment (Nashatyrev) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java:93]: `DEFAULT_DEPOSIT_SIGNATURE_VERIFIER` was intended as a default value for resetting the `depositSignatureVerifier` field after testing with disabled verifier. But you are right, it's probably an over-complication for the production code.

**Pull request merged**